### PR TITLE
Enhancements: Postgres, use /ui/api to fetch backend status, refactor to io.trino.gateway, add timeout to prepareStatement operation

### PR DESCRIPTION
**Pull request by andythsu**
- Add routing for requests to healthy backends
- Deactivate clusters with zero workers
- Add PostgreSQL support for backend database
- Rename and repackage from Presto to Trino

### CODE REVIEW

File: LICENSE [hunk 1]
```diff
@@ -1,201 +0,0 @@
-                                 Apache License
```
Comment (mosabua) [inline LICENSE:1]: Do not delete this file please

File: baseapp/pom.xml [hunk 2]
```diff
@@ -15,8 +15,8 @@
     <packaging>jar</packaging>
 
     <properties>
-        <maven.compiler.source>1.8</maven.compiler.source>
-        <maven.compiler.target>1.8</maven.compiler.target>
+        <maven.compiler.source>1.9</maven.compiler.source>
```
Comment (mosabua) [inline baseapp/pom.xml:18]: Why are we changing to Java 1.9  .. ultimately will change to 17 but is this change necessary?

File: gateway-ha/gateway-ha-config.yml [hunk 3]
```diff
@@ -4,42 +4,22 @@ routingRules:
...
-  name: prestoRouter
+  name: trinoRouter
   historySize: 1000
 
 dataStore:
-  jdbcUrl: jdbc:mysql://localhost:3306/prestogateway
-  user: root
-  password: root123
-  driver: com.mysql.cj.jdbc.Driver
+  jdbcUrl: jdbc:postgresql://localhost:5432/trino_gateway_db
```
Comment (mosabua) [inline gateway-ha/gateway-ha-config.yml:11]: Will either of these work now?

File: gateway-ha/pom.xml [hunk 4]
```diff
@@ -17,8 +17,8 @@
     <name>gateway-ha</name>
 
     <properties>
-        <maven.compiler.source>1.8</maven.compiler.source>
-        <maven.compiler.target>1.8</maven.compiler.target>
+        <maven.compiler.source>1.9</maven.compiler.source>
```
Comment (mosabua) [inline gateway-ha/pom.xml:20]: Do we need 1.9?

File: gateway-ha/pom.xml [hunk 5]
```diff
@@ -56,14 +56,14 @@
           <version>2.10.1</version>
         </dependency>
         <dependency>
-            <groupId>com.lyft.data</groupId>
+            <groupId>io.trino.gateway</groupId>
             <artifactId>baseapp</artifactId>
-            <version>${project.parent.version}</version>
+            <version>1.9.5</version>
```
Comment (mosabua) [inline gateway-ha/pom.xml:61]: Why hardcode the version here .. this seems wrong .. if anything you can set the version in dependencyManagement in the parent.

**Review by mosabua — CHANGES_REQUESTED** [review]

File: gateway-ha/pom.xml (hunk 5 above)
Comment (andythsu) [reply gateway-ha/pom.xml:61]: yes, we can just take the parent's version

File: gateway-ha/gateway-ha-config.yml (hunk 3 above)
Comment (andythsu) [reply gateway-ha/gateway-ha-config.yml:11]: yes, these should work now

File: gateway-ha/pom.xml (hunk 5 above)
Comment (andythsu) [reply gateway-ha/pom.xml:61]: ok!

File: baseapp/pom.xml (hunk 2 above)
Comment (andythsu) [reply baseapp/pom.xml:18]: this can stay 1.8, but we will need 1.9 for gateway-ha because last time it failed when I built it with 1.8.

**Comment by andythsu** [comment]
I'd like to put everything in one commit. Let me know if there are anything else to fix!

**Comment by andythsu** [comment]
> Can you please just do fixup commits and stick with the four we have
> 
> so you just do
> 
> git commmit --fixup=hashofcommittofix
> 
> and then once you have them all in
> 
> git rebase -i HEAD~5 --autosquash

I see. Np.
